### PR TITLE
feat(cms-130): integrate dashboard with real Studio runtime data

### PR DESCRIPTION
**Pull request by iipanda**
## Summary

Replaces all mock data in the Studio dashboard with real API-backed data from existing MVP contracts (schema list, content list, session, capabilities). Removes fabricated widgets that have no truthful data source in MVP.

- **Stat cards** now show real document, published, and draft counts derived from `GET /api/v1/content` pagination totals
- **Content Types** card lists schema types from `GET /api/v1/schema` with per-type document counts and published/draft progress bars
- **Recent Documents** replaces the fabricated "Recent Activity" feed with the 5 most recently updated documents from the content API
- **Removed**: "Editors online" card (live collaboration is non-MVP), weekly growth/today-updated trends (no temporal aggregation API)
- **Added**: Loading spinner, empty state with schema CTA, error state with message, forbidden state for 401/403
- **Welcome message** uses authenticated session email instead of mock user name

### New files

| File | Purpose |
|------|---------|
| `packages/studio/src/lib/content-list-api.ts` | Studio API client for `GET /api/v1/content` with filter/sort/pagination support |
| `packages/studio/src/lib/content-list-api.test.ts` | 8 tests covering query params, headers, error codes, and edge cases |
| `packages/studio/src/lib/dashboard-data.ts` | Extracted pure async dashboard data loader (testable without DOM) |
| `packages/studio/src/lib/dashboard-data.test.ts` | 9 tests covering loaded, empty, error, forbidden, type cap, and draft states |

### Acceptance criteria

1. Dashboard summary cards and lists backed by real Studio/runtime data — **done**
2. Unsupported widgets removed or replaced with truthful content — **done** (editors online, activity feed, trends)
3. Dashboard handles loading, empty, error, and forbidden states without mock fallback — **done**
4. Role-aware interaction constraints enforced — **done** (capabilities context gates schema CTA in empty state)
5. No new public contracts introduced — **done** (consumes existing APIs only)

### Dependencies

- CMS-48 (internal Studio route surfaces)
- CMS-129 (session, env context, role-aware nav)

## Test plan

- [x] `bun run check` passes (build + typecheck)
- [x] `bun run format:check` passes
- [x] 248 studio tests pass (231 existing + 17 new, 0 failures)
- [x] New tests cover: real-data state, empty state, error state (401/403/500/generic), type cap at 5, recent document fields
- [x] No studio-review impact (no dashboard fixtures exist in review app)
- [ ] Manual verification: start dev server, confirm dashboard loads real data from backend

## Summary by CodeRabbit

* **New Features**
  * Live dashboard with totals, per-type stats (capped), recent documents, and loading/forbidden/error states
  * Admin auth UX: email/password sign-in, SSO initiation, return-to handling, and redirect for unauthenticated sessions
  * New capability flag to control content-creation visibility

* **New Components**
  * Content-list API client, dashboard data loader, login API client, Login page and runtime /login route
  * Review API stubs for session, content listing, and environments

* **Tests**
  * Tests added for content-list API, dashboard loader, and login/SSO flows

* **Chores**
  * Demo seed now ensures demo RBAC grant

### CODE REVIEW

File: packages/studio/src/lib/dashboard-data.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  contentApi: StudioContentListApi,
+): Promise<DashboardLoadResult> {
+  try {
+    const [schemaTypes, totalResult, publishedResult, recentResult] =
+      await Promise.all([
+        schemaApi.list(),
+        contentApi.list({ limit: 1 }),
+        contentApi.list({ published: true, limit: 1 }),
+        contentApi.list({ sort: "updatedAt", order: "desc", limit: 5 }),
+      ]);
```
[the remark this answers is not in the document]
Comment (iipanda) [reply packages/studio/src/lib/dashboard-data.ts:46]: Is this still valid?

File: packages/studio/src/lib/runtime-ui/app/admin/page.tsx [hunk 2]
```diff
@@ -1,77 +1,206 @@
...
+
+  const statCards = [
+    {
+      label: "Documents",
+      value: data.totalDocuments,
+      icon: FileText,
+      detail:
+        data.contentTypes.length > 0
+          ? `${data.contentTypes.length} content type${data.contentTypes.length !== 1 ? "s" : ""}`
+          : undefined,
```
[the remark this answers is not in the document]
Comment (iipanda) [reply packages/studio/src/lib/runtime-ui/app/admin/page.tsx:173]: Is this still valid?

File: packages/studio/src/lib/dashboard-data.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+    const totalDocuments = totalResult.pagination.total;
+    const publishedDocuments = publishedResult.pagination.total;
+    const draftDocuments = Math.max(0, totalDocuments - publishedDocuments);
+    if (totalDocuments - publishedDocuments < 0) {
+      console.warn(
+        `Dashboard data inconsistency: published (${publishedDocuments}) exceeds total (${totalDocuments})`,
+      );
+    }
```
[the remark this answers is not in the document]
Comment (iipanda) [reply packages/studio/src/lib/dashboard-data.ts]: Is this comment still relevant?

**Comment by iipanda** [comment]
@coderabbitai resume

File: packages/studio/src/lib/dashboard-data.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (iipanda) [reply packages/studio/src/lib/dashboard-data.ts]: How could published ever be higher than total?

File: packages/studio/src/lib/runtime-ui/app/admin/page.tsx [hunk 4]
```diff
@@ -130,26 +260,19 @@ export default function DashboardPage() {
...
+          <Button
+            variant="default"
+            asChild
+            className="bg-accent hover:bg-accent-hover text-white"
+          >
+            <Link href="/admin/content">
+              <Plus className="mr-2 h-4 w-4" />
+              New Document
+            </Link>
+          </Button>
```
[the remark this answers is not in the document]
Comment (iipanda) [reply packages/studio/src/lib/runtime-ui/app/admin/page.tsx]: Is this still relevant?

File: packages/studio/src/lib/runtime-ui/app/admin/login-page.tsx [hunk 5]
```diff
@@ -0,0 +1,217 @@
+// @ts-nocheck
```
[the remark this answers is not in the document]
Comment (iipanda) [reply packages/studio/src/lib/runtime-ui/app/admin/login-page.tsx:1]: We'll address this in follow-up PR
